### PR TITLE
Be able to use Codey models on Vertex AI

### PR DESCRIPTION
**Pull request by HassanOuda**
Added the functionality to leverage 3 new Codey models from Vertex AI:
- code-bison - Code generation using the existing LLM integration
- code-gecko - Code completion using the existing LLM integration
- codechat-bison - Code chat using the existing chat_model integration

Tag maintainers/contributors who might be interested:

  - @hwchase17
  - @agola11

### CODE REVIEW

**Comment by matankley** [comment]
@HassanOuda Hi! will be merged soon? I'm really looking forward to using code chat vertex ai models using langchain.

**Comment by HassanOuda** [comment]
Need to wait for  @hwchase17 to review and approve

**Comment by vowelparrot** [comment]
This looks pretty good. Could you add an example in the notebook so it gets into our docs?

File: langchain/llms/vertexai.py [hunk 1]
```diff
@@ -50,10 +50,21 @@ def _default_params(self) -> Dict[str, Any]:
         }
         return {**base_params}
 
+    @property
```
Comment (vowelparrot) [inline langchain/llms/vertexai.py:53]: Could you update the docstrings on top_k and top_p saying they are ignored for code generation models?

**Comment by HassanOuda** [comment]
Added docs and validated it works for both.

Code chat does not support SystemMessage/Context. So added that in the docstring as well. 
@vowelparrot

**Comment by vowelparrot** [comment]
@HassanOuda was this closed intentionally?

**Comment by HassanOuda** [comment]
It was by accident while I was fixing the conflicts

**Comment by HassanOuda** [comment]
@vowelparrot created another one yesterday

**Pull request closed without merging**